src/core/shape_inference/include/slice_shape_inference.hpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <array>
+#include <openvino/op/slice.hpp>
+
+#include "slice_shape_inference_utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v8 {
+
+const std::array<std::string, 4> shape_names{"start", "stop", "step", "axes"};
```

Comment:
Yes, as it reduce memory usage in general by:
- reduce macro usage, each of add additional code as dummy copy-paste
- reduce literals size as each macro have specific literal, now is there one common shorter and input names instead of 4 specific long.
- it is repeated action for each input so not code duplication and this loop also check optional input if added to input, no dedicated `if` to check optional input.

Changed this array to `char const*` as it is simpler type and literals will be stored as raw data.
All literals have program lifetime it doesn't matter if stored in array or as literal string in validation macro(in/out function). They are in read only section and depends on OS, linker, linker script can be in persistent memory only or loaded RAM.

Moved this array to `slice` dedicated namespace as previous was in `ov::op::v8` which is too generic.